File: frontend/src/types/analysis.ts
import type { ApplicationStatus } from './job'

export interface AnalysisSummary {
  total_jobs: number
  average_match_score: number | null
  qualified_jobs: number
  contacted_jobs: number
}

export interface MatchScoreBucket {
  range: string
  count: number
}

export interface JobCategoryCount {
  category: string
  count: number
}

export interface SkillFrequency {
  skill: string
  count: number
  percentage: number
}

export interface RequirementFrequency {
  requirement: string
  count: number
}

export interface SkillCount {
  skill: string
  count: number
}

export interface AnalysisFilterOptions {
  job_categories: string[]
  application_statuses: ApplicationStatus[]
}

export interface AnalysisOverview {
  summary: AnalysisSummary
  match_score_distribution: MatchScoreBucket[]
  job_category_distribution: JobCategoryCount[]
  top_required_skills: SkillFrequency[]
  top_requirements: RequirementFrequency[]
  strength_skills: SkillCount[]
  skill_gaps: SkillCount[]
  filter_options: AnalysisFilterOptions
}

export interface AnalysisFilters {
  days?: 7 | 30
  job_category?: string
  min_score?: number
  application_status?: ApplicationStatus
}

File: backend/app/schemas/analysis.py
from typing import List, Optional

from pydantic import BaseModel, Field

from app.schemas.dashboard import ApplicationStatus


class AnalysisSummary(BaseModel):
    total_jobs: int = 0
    average_match_score: Optional[float] = None
    qualified_jobs: int = 0
    contacted_jobs: int = 0


class MatchScoreBucket(BaseModel):
    range: str
    count: int = 0


class JobCategoryCount(BaseModel):
    category: str
    count: int


class SkillFrequency(BaseModel):
    skill: str
    count: int
    percentage: float


class RequirementFrequency(BaseModel):
    requirement: str
    count: int


class SkillCount(BaseModel):
    skill: str
    count: int


class AnalysisFilterOptions(BaseModel):
    job_categories: List[str] = Field(default_factory=list)
    application_statuses: List[ApplicationStatus] = Field(default_factory=list)


class AnalysisOverviewResponse(BaseModel):
    summary: AnalysisSummary = Field(default_factory=AnalysisSummary)
    match_score_distribution: List[MatchScoreBucket] = Field(default_factory=list)
    job_category_distribution: List[JobCategoryCount] = Field(default_factory=list)
    top_required_skills: List[SkillFrequency] = Field(default_factory=list)
    top_requirements: List[RequirementFrequency] = Field(default_factory=list)
    strength_skills: List[SkillCount] = Field(default_factory=list)
    skill_gaps: List[SkillCount] = Field(default_factory=list)
    filter_options: AnalysisFilterOptions = Field(default_factory=AnalysisFilterOptions)
